tests/dialects/test_memref.py

```diff
@@ -248,7 +249,7 @@ def test_memref_cast():
     i32_memref_type = MemRefType.from_element_type_and_shape(i32, [10, 2])
     memref_ssa_value = OpResult(i32_memref_type, [], [])
 
-    res_type = MemRefType.from_element_type_and_shape(i32, [-1, -1])
+    res_type = UnrankedMemrefType.from_type(i32)
```

Comment:
Why this particular test and this aprticular line? 😅

Reply:
No reasonable reason, I guess just because that's the usecase I had in mind? :stuck_out_tongue: 
My rationale on not going further: I feel like checking that is IRDL's responsability, so should be tested extensively on that side of things, not  on each operation.
But then, I still wanted to have a case somewhere in the PR, and why a MemRefType and not an UnrankedMemrefType here :shrug:

Reply:
IRDL is actually one of the few things that we should consider well tested (or at least tested a lot).
So yeah, I agree that testing `build` functions should be useless (they are covered in `test_operation_builder`).

Reply:
Good to know! I didn't check myself yet, this is conforting :slightly_smiling_face: